docs/src/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,124 @@
+type KeyboardEvent = Event<React.KeyboardEvent<HTMLInputElement>>;
+
+interface SearchResult {
+  id: string;
+  excerpt: string;
+  title: string;
+  slug: string;
+}
+
+interface Props {
+  placeholder?: string;
+  maxWidth?: number;
+  onFocus?: InputEvent;
+  onChange?: (
+    ev: React.SyntheticEvent<HTMLInputElement>,
+  ) => (data: SearchResult[]) => void | Promise<void>;
+  onBlur?: InputEvent;
+  onSelect?: InputEvent;
+  onKeyUp?: KeyboardEvent;
+  onKeyDown?: KeyboardEvent;
+  name?: string;
+  dropdown?: boolean;
+  disabled?: boolean;
+  tabIndex?: number;
+  value?: string;
+  autoComplete?: string;
+}
+
+const Input = React.forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      placeholder = "Search for components, foundation...",
+      disabled,
+      tabIndex,
+      maxWidth,
+      dropdown = true,
+      value,
+      autoComplete,
+      onChange,
+      onFocus,
+      onBlur,
+      onKeyUp,
+      onKeyDown,
+      name,
+    },
+    ref,
+  ) => {
+    const [results, setResults] = React.useState<SearchResult[]>([]);
+    const [opened, setOpen] = React.useState(false);
+    const { index } = useSearchIndex();
+    const dropdownRef = React.useRef(null);
+    useClickOutside(dropdownRef, () => setOpen(false));
+
+    const handleSearch = (ev: React.SyntheticEvent<HTMLInputElement>) => {
+      const { value } = ev.currentTarget;
+      const indexed = Index.load(index);
+
+      setResults(
+        indexed.search(value, { expand: true }).map(({ ref }) => indexed.documentStore.getDoc(ref)),
+      );
+
+      if (results && results.length > 0) setOpen(true);
+      if (onChange) onChange(ev)(results);
+    };
+
+    return (
+      <StyledWrapper>
+        <StyledInnerWrapper maxWidth={maxWidth}>
+          <StyledInputContainer>
+            <StyledPreffix>
+              <SearchIcon />
+            </StyledPreffix>
+            <StyledInput
+              type="search"
+              placeholder={placeholder}
+              aria-label="Search through site content"
+              aria-autocomplete="list"
+              disabled={disabled}
+              onKeyUp={onKeyUp}
+              onKeyDown={onKeyDown}
+              name={name}
+              ref={ref}
+              onFocus={onFocus}
+              onBlur={onBlur}
+              onChange={handleSearch}
+              tabIndex={tabIndex}
+              autoComplete={autoComplete}
+              value={value}
+            />
+          </StyledInputContainer>
+        </StyledInnerWrapper>
+        {dropdown && results.length > 0 && opened && (
+          <StyledDropdown role="listbox" aria-label="search-dropdown" ref={dropdownRef}>
+            {results.map(({ title, excerpt, id, slug }) => (
+              <StyledDropdownItem key={id} tabIndex={-1}>
+                <Link to={slug}>{title}</Link>
+                <p>{excerpt}</p>
+              </StyledDropdownItem>
+            ))}
+          </StyledDropdown>
```

Comment:
I think it’s best to use [Downshift](https://www.downshift-js.com), or more specifically, [`useCombobox`](https://www.downshift-js.com/use-combobox). That way you get all the functionality and a11y.

Reply:
yes, downshift will be a good option, thanks